src/legacy/ui/public/chrome/directives/header_global_nav/components/header.tsx

```diff
@@ -214,6 +190,24 @@ class HeaderUI extends Component<Props, State> {
     const leftNavControls = navControls.bySide[NavControlSide.Left];
     const rightNavControls = navControls.bySide[NavControlSide.Right];
 
+    let navLinksArray = navLinks.map(navLink =>
+      navLink.hidden
+        ? null
+        : {
+            key: navLink.id,
+            label: navLink.title,
+            href: navLink.href,
+            iconType: navLink.euiIconType,
+            size: 's',
+            style: { color: 'inherit' },
+            'aria-label': navLink.title,
+            isActive: navLink.active,
+            'data-test-subj': 'appLink',
```

Comment:
This DTS also feels way to generic. Might want to think about updating it as well.